src/Jackalope/Version/VersionHandler.php

```diff
@@ -246,4 +246,60 @@ private function createFrozenNode(NodeInterface $versionNode, NodeInterface $nod
             $frozenNode->setProperty($propertyName, $property->getValue());
         }
     }
+
+    public function restoreItem($removeExisting, $versionPath, $path)
+    {
+        $node = $this->objectManager->getNodeByPath($path);
+
+        if ($node->isModified()) {
+            throw new InvalidItemStateException(
+                sprintf(
+                    'Node "%s" contains unsaved changes',
+                    $path
+                )
+            );
+        }
+
+        $versionNode = $this->objectManager->getNodeByPath($versionPath, 'Version\Version');
+        $frozenNode = $versionNode->getNode('jcr:frozenNode');
+
+        // TODO reset primary type
+
+        if ($frozenNode->hasProperty('jcr:frozenMixinTypes')) {
+            $node->setProperty('jcr:mixinTypes', $frozenNode->getPropertyValue('jcr:frozenMixinTypes'));
+        }
+
+        // TODO reset uuid
+
+        foreach ($frozenNode->getProperties() as $property) {
+            /** @var PropertyInterface $property */
+            $propertyName = $property->getName();
+            if ($propertyName == 'jcr:frozenPrimaryType'
+                || $propertyName == 'jcr:frozenMixinTypes'
+                || $propertyName == 'jcr:frozenUuid'
+            ) {
+                continue;
+            }
+
+            $nodeProperty = $node->getProperty($propertyName);
```

Comment:
it could be that the property has been deleted meanwhile. then you would need to create a new property, using the value and the type of the frozen property.